qcodes/instrument_drivers/AlazarTech/ATS.py

```diff
@@ -445,6 +445,40 @@ def acquire(self, mode=None, samples_per_record=None,
         samples_per_record = self.samples_per_record.raw_value
         records_per_buffer = self.records_per_buffer.raw_value
 
+        # bits per sample
+        _, bits_per_sample = self.api.get_channel_info_(self._handle)
+
+        # channels
+        channels_binrep = self.channel_selection.raw_value
+        number_of_channels = self.get_num_channels(channels_binrep)
+
+        # In the following we need to consider the size of the buffer
+        # in two different scenarios as several Alazar cards have sample sizes
+        # that are in fractions of bytes. (such as 12 bits).
+        # We are transferring data padded to
+        # whole bytes. I.e a sample of 12 bits will take up 16 bits when
+        # transferred so we are allocating buffers of that size.
+        # However, when calculating internal limitations on the card we are
+        # using the fractional sizes of samples
+
+        # number of bytes per sample rounded up to the nearest integer
+        whole_bytes_per_sample = (bits_per_sample + 7) // 8
+        transfer_record_size = whole_bytes_per_sample * samples_per_record
+        transfer_buffer_size = (transfer_record_size *
+                                records_per_buffer * number_of_channels)
+
+        sample_type = (
+            ctypes.c_uint16 if whole_bytes_per_sample > 1 else ctypes.c_uint8)
+
+        internal_buffer_size_requested = (bits_per_sample * samples_per_record *
+                                          records_per_buffer) // 8
+
+        if internal_buffer_size_requested > max_buffer_size:
+            raise RuntimeError(f"Requested a buffer of size: "
+                               f"{internal_buffer_size_requested / 1024 ** 2}"
+                               f" MB. The maximum supported size is "
+                               f"{max_buffer_size / 1024 ** 2} MB.")
```

Comment:
How about also adding the recommended here as well?
```suggestion
                               f"{max_buffer_size / 1024 ** 2} MB "
                               f"(recommended is <8MB).")
```